### PR TITLE
fix!: handle non-inertia `location` redirects

### PR DESCRIPTION
**Pull request by innocenzi**
Sometimes, we want to perform an external redirect that can be initiated on the first visit or on an Inertia visit.

In such a case, Inertia's `location` won't work because it will only simply return a normal `409` response with the `X-Inertia-Location`, but since the front-end is not loaded the end result will be a blank page. 

To fix this, we need to check if the current request is an Inertia one, perform the `location` redirect, or return a classic `RedirectResponse` otherwise. 

This PR fixes that issue by performing that check internally so the developer doesn't have to.

This is *technically* a breaking change since I had to modify the original test, but it's a breaking change that is more likely to fix issues than to create some.

---

TL;DR: with this PR, instead of doing:

```php
if (request()->inertia()) {
    return inertia()->location($url);
}

return redirect()->to($url);
```

We can just do:
```php
return inertia()->location($url);
```

---

Fixes https://github.com/inertiajs/inertia-laravel/issues/295

### CODE REVIEW

**Comment by narasimhajupally** [comment]
Hello, I just checked the code changes for this issue and tried to do similar in my controller
```
if (Request::inertia()) {
       return Inertia::location(tenantRoute($email->domain->domain, 'home'));
}
return redirect(tenantRoute($email->domain->domain, 'home'));
```
I am getting this exception
`ErrorException Cannot unbind $this of closure using $this`
any idea ? 
Thanks.

**Comment by claudiodekker** [comment]
Thanks @innocenzi!
